cachi2/core/package_managers/rpm/main.py

```diff
@@ -0,0 +1,204 @@
+        ]
+        license = run_cmd(cmd=["rpm", *rpm_args], params={})
+
+        with rpmfile.open(file_path) as rpm:
+            vendor = rpm.headers.get("vendor", b"").decode()
+            vendor = quote(vendor.lower())
+
+            name = rpm.headers.get("name", b"").decode()
+            version = rpm.headers.get("version", b"").decode()
+            release = rpm.headers.get("release", b"").decode()
+            arch = rpm.headers.get("arch", b"").decode()
+            download_url = quote(file_metadata["url"])
+
+            # https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#rpm
+            # https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#known-qualifiers-keyvalue-pairsa
+            purl = (
+                f"pkg:rpm{'/' if vendor else ''}{vendor}/{name}@{version}{release}"
+                f"?arch={arch}{'&epoch=' if epoch else ''}{epoch}&download_url={download_url}"
+            )
+
+            components.append(
+                Component(
+                    name=name,
+                    version=version,
+                    purl=purl,
+                    # NOTE: add checksum into SBOM?
+                )
+            )
+
+
+def process_packages(path: Path) -> None:
+    """
+    Search structure for all repoid dirs and create repository metadata
+    out of its RPMs. Skip sources dir that contains only SRPMs.
+    """
+    package_dir = RootedPath(path).join_within_root(DEFAULT_PACKAGE_DIR)
+    for entry in os.listdir(package_dir):
+        if entry == "sources" or not os.path.isdir(os.path.join(package_dir, entry)):
+            continue
+        arch = entry
+        for entry in os.listdir(os.path.join(package_dir, arch)):
+            if not os.path.isdir(os.path.join(package_dir, arch, entry)):
+                continue
+            repoid = entry
+            repodir = os.path.join(package_dir, arch, repoid)
+            createrepo(repoid, repodir)
+
+
+def createrepo(repoid: str, repodir: str) -> None:
+    log.info(f"Creating repository metadata for repoid '{repoid}': {repodir}")
+    cmd = ["/usr/bin/createrepo", repodir]
+    log.debug("$ " + shlex.join(cmd))
+    stdout = run_cmd(cmd, params={})
+    log.debug(stdout)
+
+
+def _generate_repofiles(path: RootedPath) -> list[ProjectFile]:
+    """
+    Search structure and generate repofile for each arch in its dir.
+    Repofile contains all arch's repoids. Skip sources dir that contains only SRPMs.
+    """
+    project_files: list[ProjectFile] = []
+    package_dir = path.join_within_root(DEFAULT_PACKAGE_DIR)
+    for subdir in os.listdir(package_dir):
+        if subdir == "sources":
+            continue
+        arch = subdir
+        log.debug(f"Preparing repofile for arch '{arch}'")
+        abspath = os.path.join(package_dir, arch, "cachi2.repo")
+        template = ""
+        for repoid in os.listdir(os.path.join(package_dir, arch)):
+            localpath = os.path.join(package_dir, arch, repoid)
```

Comment:
This won't work because you set this to a fixed path that won't exist in the container. Since you can't have relative paths in the `file` URI protocol, you'll need to make use of our `--for-output-dir` CLI option, have a look at other places where this is used.

Reply:
I just noticed that in the tests was doing with hermetic builds, I accidentally was mounting the output dir in a way that it matched what I had in my local environment, that's why the builds succeeded. 

That is not the case in my early draft of the integration tests, and I'm getting failures such as:
```
STEP 2/3: RUN microdnf -y install 	vim
Updating and loading repositories:
updates-released                       ???% |   0.0   B/s |  -1.0   B |  ?     
 >>> Curl error (37): Couldn't read a file:// file for file:///tmp/pytest-of-bpim
```